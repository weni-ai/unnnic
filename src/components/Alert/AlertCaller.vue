<template>
    <button @click="callAlert"> Launch alert </button>
</template>

<script>
import alert from '../../utils/call';

export default {
  props: {
    title: {
      type: String,
      default: null,
    },
    text: {
      type: String,
      default: null,
    },
    icon: {
      type: String,
      default: null,
    },
    enabled: {
      type: Boolean,
      default: true,
    },
    scheme: {
      type: String,
      default: null,
    },
    seconds: {
      type: Number,
      default: 3,
    },
    closeText: {
      type: String,
      default: null,
    },
    position: {
      type: String,
      default: 'top-right',
    },
  },
  methods: {
    callAlert() {
      alert.callAlert({ props: this.$props, seconds: this.seconds });
    },
  },
};
</script>
