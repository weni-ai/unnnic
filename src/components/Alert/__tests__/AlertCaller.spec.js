import { mount } from '@vue/test-utils';
import { describe, expect, test, vi, beforeEach, afterEach } from 'vitest';
import AlertCaller from '@/components/Alert/AlertCaller.vue';
import alert from '@/utils/call';

vi.mock('@/utils/call', () => ({
  default: {
    callAlert: vi.fn(),
  },
}));

describe('AlertCaller.vue', () => {
  let wrapper;

  const defaultProps = {
    title: 'Test Title',
    text: 'Test Text',
    icon: 'test-icon',
    enabled: true,
    scheme: 'primary',
    seconds: 5,
    closeText: 'Close',
    position: 'top-right',
  };

  beforeEach(() => {
    wrapper = mount(AlertCaller, {
      props: defaultProps,
    });
  });

  afterEach(() => {
    wrapper.unmount();
    vi.clearAllMocks();
  });

  test('renders correctly', () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.find('button').text()).toBe('Launch alert');
  });

  test('calls callAlert method when button is clicked', async () => {
    const callAlertSpy = vi.spyOn(wrapper.vm, 'callAlert');
    await wrapper.find('button').trigger('click');
    expect(callAlertSpy).toHaveBeenCalledTimes(1);
  });

  test('callAlert method invokes alert.callAlert with correct props', async () => {
    await wrapper.find('button').trigger('click');
    expect(alert.callAlert).toHaveBeenCalledWith({
      props: defaultProps,
      seconds: defaultProps.seconds,
    });
  });

  test.each([
    ['title', 'New Title'],
    ['text', 'New Text'],
    ['icon', 'new-icon'],
    ['enabled', false],
    ['scheme', 'secondary'],
    ['seconds', 10],
    ['closeText', 'Dismiss'],
    ['position', 'bottom-left'],
  ])('updates %s prop correctly', async (propName, newValue) => {
    await wrapper.setProps({ [propName]: newValue });
    await wrapper.find('button').trigger('click');
    const expectedProps = { ...defaultProps, [propName]: newValue };
    expect(alert.callAlert).toHaveBeenCalledWith({
      props: expectedProps,
      seconds: expectedProps.seconds,
    });
  });

  test('passes default values when props are not provided', async () => {
    wrapper = mount(AlertCaller, {
      props: {},
    });
    await wrapper.find('button').trigger('click');
    expect(alert.callAlert).toHaveBeenCalledWith({
      props: {
        title: null,
        text: null,
        icon: null,
        enabled: true,
        scheme: null,
        seconds: 3,
        closeText: null,
        position: 'top-right',
      },
      seconds: 3,
    });
  });

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
