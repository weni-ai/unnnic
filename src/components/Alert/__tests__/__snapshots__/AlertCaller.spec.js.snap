// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlertCaller.vue > matches snapshot 1`] = `"<button>Launch alert</button>"`;
